Source/Core/VideoBackends/OGL/SamplerCache.cpp

```diff
@@ -15,141 +15,110 @@ namespace OGL
 {
 std::unique_ptr<SamplerCache> g_sampler_cache;
 
-SamplerCache::SamplerCache() : m_last_max_anisotropy()
+SamplerCache::SamplerCache()
 {
-  glGenSamplers(2, m_sampler_id);
-  glSamplerParameteri(m_sampler_id[0], GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-  glSamplerParameteri(m_sampler_id[0], GL_TEXTURE_MAG_FILTER, GL_NEAREST);
-  glSamplerParameteri(m_sampler_id[0], GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
-  glSamplerParameteri(m_sampler_id[0], GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-  glSamplerParameteri(m_sampler_id[1], GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-  glSamplerParameteri(m_sampler_id[1], GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-  glSamplerParameteri(m_sampler_id[1], GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
-  glSamplerParameteri(m_sampler_id[1], GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glGenSamplers(1, &m_point_sampler);
+  glGenSamplers(1, &m_linear_sampler);
+  glSamplerParameteri(m_point_sampler, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glSamplerParameteri(m_point_sampler, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glSamplerParameteri(m_point_sampler, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glSamplerParameteri(m_point_sampler, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glSamplerParameteri(m_linear_sampler, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+  glSamplerParameteri(m_linear_sampler, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+  glSamplerParameteri(m_linear_sampler, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glSamplerParameteri(m_linear_sampler, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
 }
 
 SamplerCache::~SamplerCache()
 {
   Clear();
-  glDeleteSamplers(2, m_sampler_id);
+  glDeleteSamplers(1, &m_point_sampler);
+  glDeleteSamplers(1, &m_linear_sampler);
 }
 
 void SamplerCache::BindNearestSampler(int stage)
 {
-  glBindSampler(stage, m_sampler_id[0]);
+  glBindSampler(stage, m_point_sampler);
 }
 
 void SamplerCache::BindLinearSampler(int stage)
 {
-  glBindSampler(stage, m_sampler_id[1]);
+  glBindSampler(stage, m_linear_sampler);
 }
 
-void SamplerCache::SetSamplerState(int stage, const TexMode0& tm0, const TexMode1& tm1,
-                                   bool custom_tex)
+void SamplerCache::SetSamplerState(u32 stage, const SamplerState& state)
 {
-  // TODO: can this go somewhere else?
-  if (m_last_max_anisotropy != g_ActiveConfig.iMaxAnisotropy)
-  {
-    m_last_max_anisotropy = g_ActiveConfig.iMaxAnisotropy;
-    Clear();
-  }
+  if (m_active_samplers[stage].first == state && m_active_samplers[stage].second != 0)
+    return;
 
-  Params params(tm0, tm1);
-
-  // take equivalent forced linear when bForceFiltering
-  if (g_ActiveConfig.bForceFiltering)
+  auto it = m_cache.find(state);
+  if (it != m_cache.end())
```

Comment:
Small nitpick, but I'd prefer it the other way:
`if (it == m_cache.end()) {
  GLuint sampler;
  glGenSamplers(1, &sampler);
  SetParameters(sampler, state);
  it = m_cache.emplace(state, sampler);
}
m_active_samplers[stage].first = state;
m_active_samplers[stage].second = it->second;
glBindSampler(stage, it->second);`